File: src/ClientApp/src/app/model/produced-product.ts
import { ConfiguredProduct } from './configured-product';
import { ProductPart } from './product-part';

export interface ProducedProduct {
  id?: number;
  configuredProductId: number;
  configuredProductName: string;
  aasId: string;
  globalAssetId: string;
  productionDate: Date;
  bestandteile: ProductPart[];

  // Optional für erweiterte Anzeige
  configuredProduct?: ConfiguredProduct;
}

File: src/Database/Model/ProducedProduct.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace AasDemoapp.Database.Model
{
    public class ProducedProduct : ISoftDelete
    {
        public long? Id { get; set; }
        public ConfiguredProduct ConfiguredProduct { get; set; } = null!;
        public long ConfiguredProductId { get; set; }
        public List<ProductPart> Bestandteile { get; set; } = [];
        public string AasId { get; set; } = string.Empty;
        public string GlobalAssetId { get; set; } = string.Empty;

        public DateTime ProductionDate { get; set; }
        public bool IsDeleted { get; set; }
        public DateTime? DeletedAt { get; set; }

        public double PCFValue { get; set; }
    }
}
